### ISSUE TITLE
Locale/currency pair rhg-Rohg_MM/PAB crashes.

### ISSUE THREAD

**Issue by oscbyspro** [oscbyspro/DiffableTextViews#23]
This is because the current implementation assumes that all formatting characters are unique, but in this case the localized currency expression contains a fraction separator. 

```swift
func testVirtualCharactersAreNotAlwaysUnique() {
    let number = -1234567.89
    let currencyCode = "PAB"
    let locale = Locale(identifier: "rhg-Rohg_MM")
    let formatted = number.formatted(.currency(code: currencyCode).locale(locale))
    XCTAssertEqual(formatted, "-B/. 1,234,567.89") // currency contains fraction separator
}
```

**Comment by oscbyspro** [comment]
I don't think this will take long to solve, maybe today or tomorrow. I'm just upset that people would do this...

**Comment by oscbyspro** [comment]
Solved by: bf6d33a1863aee139272d73a49d20c2ee7d2203a.
